src/map/utils/battleutils.cpp

```diff
@@ -2519,6 +2519,9 @@ namespace battleutils
             // Do we get TP for damaging spells?
             int16 tp = battleutils::CalculateSpellTP(PAttacker, PSpell);
             PAttacker->addTP(tp);
+
+            // Targets of damaging spells gain 50 tp
+            PDefender->addTP((uint16)50);
```

Comment:
This cast is misleading, and most likely unnecessary.  `addTP()`'s parameter is a signed int16, not unsigned

Reply:
that  makes sense. i had borrowed code from [battleutils.cpp#L2327](https://github.com/LandSandBoat/server/blob/219a712da21f650eddf07ac3d7b0eaa990d9ef96/src/map/utils/battleutils.cpp#L2327) so i just went with what was already used. Even in the start of that ```if statement``` the used int16.